lib/whitehall_importer/migrate_assets.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module WhitehallImporter
+  class MigrateAssets
+    attr_reader :whitehall_import
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(whitehall_import)
+      @whitehall_import = whitehall_import
+    end
+
+    def call
+      whitehall_import.assets.reject { |asset| asset.state == "processed" }.each do |whitehall_asset|
+        begin
+          whitehall_asset.update!(state: "processing")
+          if whitehall_asset.main_asset.state == "live" &&
+              (whitehall_asset.variant == "original" || whitehall_asset.image_revision.present?)
+            GdsApi.asset_manager.update_asset(
+              whitehall_asset.asset_manager_id,
+              redirect_url: whitehall_asset.main_asset.file_url,
+            )
+          else
+            GdsApi.asset_manager.delete_asset(whitehall_asset.asset_manager_id)
+          end
+          whitehall_asset.update!(state: "processed")
+        rescue GdsApi::BaseError, StandardError => e
+          whitehall_asset.update!(state: "migration_failed", error_message: e.inspect)
+        end
+      end
+    end
```

Comment:
I think there's a bit too much going on in this method and could used being separated out.